### PR TITLE
apk artifacts analyzer closes#2444 + upgraded stringsifter

### LINKED ISSUE #2444: [Analyzer] APK_Artifacts
## Name
APK_Artifacts

## Link
https://github.com/guelfoweb/artifacts

## Type of analyzer
file anayzer for APKs only

## Why should we use it
cool open source tool to triage apk packages

## Possible implementation
https://github.com/guelfoweb/artifacts with all the possible options

### PR DESCRIPTION
**Pull request by g4ze**
closes #2444

# Description

Please include a summary of the change and link to the related issue.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ ] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [ ] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ ] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require any API key), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
    - [ ] If the plugin interacts with an external service, I have created an attribute called precisely `url` that contains this information. This is required for Health Checks. 
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have added that raw JSON sample to the `MockUpResponse` of the `_monkeypatch()` method. This serves us to provide a valid sample for testing.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [ ] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

**Comment by g4ze** [comment]
```
malware_tools_analyzers-user@0e3f29973277:/opt/deploy/artifacts/artifa
cts$ /opt/deploy/artifacts/venv/bin/python3 /opt/deploy/artifacts/artifacts/artifacts.py ./bitbar-sample-app.apk 
{
    "version": "1.1.1",
    "md5": "00cc5435151aa38a091781922c0390a4",
    "dex": [
        "classes.dex"
    ],
    "library": [],
    "network": {
        "ip": [],
        "url": [],
        "param": []
    },
    "root": [],
    "string": {
        "base64": [],
        "telegram_id": [],
        "known": []
    },
    "family": {
        "name": "CryCrypto",
        "match": 20.63,
        "value": {
            "permission": 28.57,
            "application": 0.0,
            "intent": 33.33
        }
    },
    "sandbox": [
        "https://tria.ge/s?q=00cc5435151aa38a091781922c0390a4",
        "https://www.joesandbox.com/analysis/search?q=00cc5435151aa38a091781922c0390a4",
        "https://www.virustotal.com/gui/search/00cc5435151aa38a091781922c0390a4",
        "https://bazaar.abuse.ch/browse.php?search=md5:00cc5435151aa38a091781922c0390a4",
        "https://koodous.com/apks?search=00cc5435151aa38a091781922c0390a4"
    ],
    "elapsed_time": 0.09
}

```
the command works perfectly fine inside the container but fails while analyzer run:
image

File: api_app/analyzers_manager/file_analyzers/artifacts.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+from tests.mock_utils import MockUpResponse
+
+logger = logging.getLogger(__name__)
+
+
+class Artifacts(FileAnalyzer, DockerBasedAnalyzer):
+    name: str = "apk_analyzer"
+    url: str = "http://malware_tools_analyzers:4002/artifacts"
+    # interval between http request polling
+    poll_distance: int = 5
```
Comment (mlodic) [inline api_app/analyzers_manager/file_analyzers/artifacts.py:14]: keep it 2

**Review by mlodic — COMMENTED** [review]
I think the 2 calls (cli and gui) are different. In the gui the tool uses the similarity and I think it would work only with another sample compared.

Does the "report" option produce something different completely? Cause, in the `cli` option, it seems that it's adding new elements in the json report. In that case, it should be included always in the results and you could avoid those parameters that choose the type of the analysis

**Comment by g4ze** [comment]
How is gui and cli exec different?
Are we not literally running the same command internally?
The same command gives desired op when I shell inside the container.

**Comment by g4ze** [comment]
{
            "name": "APK_Artifacts",
            "process_time": 5.07,
            "status": "SUCCESS",
            "end_time": "2024-08-27T10:03:15.563886Z",
            "parameters": {},
            "type": "analyzer",
            "id": 72,
            "report": {
                "dex": [
                    "classes.dex"
                ],
                "md5": "8a05a189e58ccd7275f7ffdf88c2c191",
                "root": [],
                "family": {
                    "name": "CryCrypto",
                    "match": 11.11,
                    "value": {
                        "intent": 33.33,
                        "permission": 0.0,
                        "application": 0.0
                    }
                },
                "string": {
                    "known": [],
                    "base64": [],
                    "telegram_id": []
                },
                "library": [],
                "network": {
                    "ip": [],
                    "url": [],
                    "param": []
                },
                "sandbox": [
                    "https://tria.ge/s?q=8a05a189e58ccd7275f7ffdf88c2c191",
                    "https://www.joesandbox.com/analysis/search?q=8a05a189e58ccd7275f7ffdf88c2c191",
                    "https://www.virustotal.com/gui/search/8a05a189e58ccd7275f7ffdf88c2c191",
                    "https://bazaar.abuse.ch/browse.php?search=md5:8a05a189e58ccd7275f7ffdf88c2c191",
                    "https://koodous.com/apks?search=8a05a189e58ccd7275f7ffdf88c2c191"
                ],
                "version": "1.1.1",
                "elapsed_time": 0.02
            },
            
            
image

File: api_app/analyzers_manager/migrations/0119_analyzer_config_apk_artifacts.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    "description": "[Artifacts](https://github.com/guelfoweb/artifacts) is a tool that does APK strings analysis. Useful for first analysis.",
+    "disabled": False,
+    "soft_time_limit": 60,
+    "routing_key": "default",
+    "health_check_status": True,
+    "type": "file",
+    "docker_based": False,
+    "maximum_tlp": "RED",
+    "observable_supported": [],
+    "supported_filetypes": ["application/vnd.android.package-archive"],
```
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0119_analyzer_config_apk_artifacts.py:25]: I would set the same mimetypes that are used for APKiD - 

application/java-archive
application/vnd.android.package-archive
application/x-dex
application/zip

**Review by mlodic — COMMENTED** [review]
last thing: can you try 2 runs of StringsInfo to understand whether it still works after the upgrade?

1st run normal
2nd run with option `rank_strings` active

ty

**Comment by g4ze** [comment]
image
image

**Comment by g4ze** [comment]
all gtg
docs here https://github.com/intelowlproject/docs/pull/4

**Comment by g4ze** [comment]
Wohoooo all tasks done!!!!!!🤝

**Pull request merged**